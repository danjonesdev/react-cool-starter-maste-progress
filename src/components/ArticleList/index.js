import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import b from '../../theme/bootstrap-grid.min.css';
import s from './styles.scss';

type Props = { list: Object };

export class ArticleList extends PureComponent {
  props: Props;

  static defaultProps: {
    ArticleList: {
      test: null,
    }
  };

  render() {
    return (
    <div className={s.ArticleList}>
      <h4>Popular this week</h4>
      <div className={s.grid}>
        {this.props.list.map(article => (
            <div key={article.id} className={s.col}>
              <Link to={`/ArticleInfo/${article.id}`}>
                <div className={s.card}>
                  <div className={s.img}>
                    <img src="https://qph.ec.quoracdn.net/main-thumb-t-7187-200-WTlZyW922iBQFzeQKR99N4bZ4w44Drp2.jpeg" alt="" />
                  </div>
                  <p className={s.title}>{article.title}</p>
                  </div>
              </Link>
            </div>
        ))}
      </div>
    </div>
    );
  }
}

ArticleList.defaultProps = {
  list: {
    id: '',
    title: '',
  },
};

export default ArticleList;
