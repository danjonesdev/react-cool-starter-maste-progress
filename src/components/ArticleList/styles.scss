@import '../../theme/variables';
@import url( 'https://fonts.googleapis.com/css?family=Ubuntu');
@import url( 'https://fonts.googleapis.com/css?family=Open+Sans:600');

.ArticleList {
  .grid {
    max-width: 1200px;
    margin: 0 auto;
    text-align: center;
  }
  @media all and (max-width: 1200px) {
    .grid {
      max-width: 800px;
    }
  }

  .col {
    width: 250px;
    height: 250px;
    display: inline-block;
    padding: 10px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    opacity: 0;
    animation: fadeIn 0.5s ease both;
    margin-bottom: 45px;
  }
  @media all and (max-width: 1200px) {
    .col {
      width: 200px;
      height: 200px;
    }
  }

  .col-lg {
    width: 300px;
    height: 300px;
    display: inline-block;
    padding: 10px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  @media all and (max-width: 800px) {
    .col-lg {
      width: 200px;
      height: 200px;
    }
  }
  @keyframes fadeIn {
    0% {
      opacity: 0;
    }

    100% {
      opacity: 1;
    }
  }

  a {
    text-decoration: none;
  }

  .card {
    width: 100%;
    height: 100%;
    position: relative;

    .img {
      width: 100%;
      height: 100%;
      overflow: hidden;

      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
        transition: all 1s ease;

        &:hover {
          transform: scale(1.1);
          -ms-filter:  "progid:DXImageTransform.Microsoft.Matrix(M11=1.5, M12=0, M21=0, M22=1.5, SizingMethod='auto expand')";
          filter: progid:DXImageTransform.Microsoft.Matrix(M11=1.5, M12=0, M21=0, M22=1.5, SizingMethod=  'auto expand');
        }
      }
    }

    .title {
      position: absolute;
      font-size: 13px;
      line-height: 18px;
      color: #1a1a1a;
      text-align: left;
      font-weight: bold;
      left: 5px;
      width: -webkit-calc(100% - 5px);
      width: -moz-calc(100% - 5px);
      width: calc(100% - 5px);
      top: -webkit-calc(100% - 2px);
      top: -moz-calc(100% - 2px);
      top: calc(100% - 2px);
    }
  }
}
