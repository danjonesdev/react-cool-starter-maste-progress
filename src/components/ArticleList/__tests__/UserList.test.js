import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import ArticleList from '../index';

describe('<ArticleList />', () => {
  test('renders', () => {
    const mockData = [{ id: '1', name: 'Welly' }];
    const tree = renderer.create(
      <StaticRouter location={''} context={{}}>
        <ArticleList list={mockData} />
      </StaticRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
