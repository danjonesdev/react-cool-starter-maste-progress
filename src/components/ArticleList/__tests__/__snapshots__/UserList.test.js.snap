// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> renders 1`] = `
<div
  className={undefined}
>
  <h4>
    Article List
  </h4>
  <ul>
    <li>
      <a
        href="/ArticleInfo/1"
        onClick={[Function]}
      >
        Welly
      </a>
    </li>
  </ul>
</div>
`;
